### ISSUE TITLE
Network errors are not fired to ConnectionRequest.handleException nor Socket.connectionError on network down or timeout on iPhone

### ISSUE THREAD

**Issue by stoiev** [codenameone/CodenameOne#1535]
When network is down, seems that iOS port does not provide any report to ConnectionRequest

Steps to reproduce:

1. On KitchenSink demo, do little changes on src/codename1/demos/kitchen/WebServices.java (diff bellow):
  *  setFailSilently(false) on WebServiceRequest constructor
  *  Overwride handleException with Dialog.show("Network error", err.getMessage(), "Ok", "Error");
2. Build code to a iPhone5s or iPod and turn down any network connection (wifi or APN)
3. Try to use Web Service demo. First, it blinks the ProgressListener, after that,  ProgressListener is shown and never get down.

This same code on simulator and android works like a charm (it shows the dialog with the Exception). 

```diff
192d191
<             setFailSilently(false);
213,218d211
<         
<         @Override
<         protected void handleException(Exception err){
<             Dialog.show("Network error", err.getMessage(), "Ok", "Error");
<         }
<         
```

I've tried to catch this same Exception with NetworkManager.addErrorListener and its not fired either. Here is some logs console logs of this problem:
```
Jun 17 15:34:49 iPod-Touch-Fabiok Escalones[2350] <Warning>: [EDT] 0:0:0,383 - Start
Jun 17 15:34:49 iPod-Touch-Fabiok backboardd[26] <Warning>: CoreAnimation: timed out fence 13dfb
Jun 17 15:34:49 iPod-Touch-Fabiok kernel[0] <Debug>: launchd[2351] Builtin profile: PasteBoard (sandbox)
Jun 17 15:34:49 iPod-Touch-Fabiok networkd[67] <Error>: client_connection_cellular_start no interface match
Jun 17 15:34:49 iPod-Touch-Fabiok networkd[67] <Error>: client_connection_cellular_start no interface match
Jun 17 15:34:51 iPod-Touch-Fabiok Escalones[2350] <Warning>: [EDT] 0:0:3,37 - Show Main UI
Jun 17 15:34:57 iPod-Touch-Fabiok networkd[67] <Error>: client_connection_cellular_start no interface match
Jun 17 15:34:57 iPod-Touch-Fabiok networkd[67] <Error>: client_connection_cellular_start no interface match
Jun 17 15:34:57 iPod-Touch-Fabiok Escalones[2350] <Warning>: [Network Thread] 0:0:8,503 - The Internet connection appears to be offline.
Jun 17 15:34:57 iPod-Touch-Fabiok Escalones[2350] <Warning>: java.io.IOException
    at com_codename1_io_NetworkManager_NetworkThread.run:300
    at com_codename1_impl_CodenameOneThread.run:176
    at java_lang_Thread.runImpl:153
Jun 17 15:34:57 iPod-Touch-Fabiok Escalones[2350] <Warning>: 
Jun 17 15:34:57 iPod-Touch-Fabiok Escalones[2350] <Warning>: [Network Thread] 0:0:8,520 - java.io.IOException: The Internet connection appears to be offline.
Jun 17 15:34:57 iPod-Touch-Fabiok Escalones[2350] <Warning>: [Network Thread] 0:0:8,557 - java.io.IOException: The Internet connection appears to be offline.
Jun 17 15:34:57 iPod-Touch-Fabiok Escalones[2350] <Warning>: Deleting thread 3
Jun 17 15:37:14 iPod-Touch-Fabiok com.apple.launchd[1] (com.apple.timed[1234]) <Notice>: (com.apple.timed) Idle-exit job was jettisoned. Will bypass throttle interval for next on-demand launch.
Jun 17 15:37:14 iPod-Touch-Fabiok com.apple.launchd[1] (com.apple.timed[1234]) <Notice>: (com.apple.timed) Exited: Killed: 9
Jun 17 15:37:14 iPod-Touch-Fabiok com.apple.launchd[1] (UIKitApplication:com.TapMediaLtd.BarcodeReader[0xd061][1159]) <Notice>: (UIKitApplication:com.TapMediaLtd.BarcodeReader[0xd061]) Exited: Killed: 9
```

On my own application, from a timeout, it was raised a crash report dialog, but no exception was fired to handleException. Here is the log:
```
Jun 17 21:23:08 iPod-Touch-Fabiok Escalones[4367] <Warning>: [Network Thread] 0:20:39,45 - The request timed out.
Jun 17 21:23:08 iPod-Touch-Fabiok Escalones[4367] <Warning>: java.io.IOException
    at com_codename1_io_NetworkManager_NetworkThread.run:290
    at com_codename1_impl_CodenameOneThread.run:176
    at java_lang_Thread.runImpl:153
Jun 17 21:23:08 iPod-Touch-Fabiok Escalones[4367] <Warning>: 
Jun 17 21:23:08 iPod-Touch-Fabiok Escalones[4367] <Warning>: [Network Thread] 0:20:39,79 - java.io.IOException: The request timed out.
Jun 17 21:23:08 iPod-Touch-Fabiok Escalones[4367] <Warning>: [Network Thread] 0:20:39,88 - java.io.IOException: The request timed out.
```

**Comment by stoiev** [comment]
After same investigation, I've be able to capture and deal with this errors setting `Display.getInstance().setCrashReporter` with a custom CrashReporter. Here it is:

```java
public class CrashReporter implements CrashReport {

    public void exception(final Throwable t) {
        Thread actual = Thread.currentThread();
        if (actual.getName().equals("Network Thread")) {
            Display.getInstance().callSerially(new Runnable() {
                public void run() {
                    Log.p("Network error catched up");
                    Dialog.show("Network error", t.getMessage(), "OK", "Cancel");
                }
            });
        }
    }
}

```
You guys could confirm it much better, but it seems that the 'automatic' mode of iOS SDK Network stack only generate this kind of report on a 'general error listener' fashion, so it is difficult to relate these error with the original ConnectionRequest. I'm I rigth or talking nonsense?

My dummy java point of view let me think that If NetworkThread class was public visible, I could restore the context from this point, but i'm just guessing.

The context is important on some cases, like removing ConectionRequest's attached progress listeners or requests created with `setFailSilently(true);` (that will not affect application on one or another failed requests)  

At the moment, I will try to create a universal form property to deal with displayed progress listeners (something like `Display.getInstance().getCurrent().getPropertyValue('listener');`)

**Comment by codenameone** [comment]
So to be clear, timeout isn't firing an error?

**Comment by stoiev** [comment]
Seems that network exception in general are not fired to ```handleException```
I'm showing a Dialog on my custom CrashReporter with the Exception message and I could get messages informing network unavaiability ("The Internet connection appears to be offline"), timeout ("The request timed out"), and host not found.

**Comment by stoiev** [comment]
This issue happens on 'lastest' Codename One version.
On 3.0 the iOS and Android works the same (throwing exceptions on ConnectionRequest.handleException)
It's ok to raise this kind of issues here? (For future reference)

**Comment by codenameone** [comment]
This is a probably a regression due to the fix of #1519 which would be pretty tough to fix. I'll look into that.

**Comment by codenameone** [comment]
We fixed the relevant regression, please let us know if this is still happening.
